Nodejs/Tests/AnalysisTests/AnalysisFile.cs

```diff
@@ -131,6 +131,20 @@ static class Analysis {
             List<AnalysisFile> files = new List<AnalysisFile>();
             foreach (var file in Directory.GetFiles(directory, "*", SearchOption.AllDirectories)) {
                 if (String.Equals(Path.GetExtension(file), ".js", StringComparison.OrdinalIgnoreCase)) {
+                    var relativeFile = file.Substring(directory.Length);
+                    int nestedModulesCount = 0;
+                    int startIndex = 0;
+                    int index = relativeFile.IndexOf("node_modules", startIndex, StringComparison.OrdinalIgnoreCase);
+                    while (index != -1) {
+                        nestedModulesCount++;
+                        if (nestedModulesCount > limits.NestedModulesLimit) {
+                            continue;
```

Comment:
the check is nested now, so continue is going to continue the while loop, not the outer for loop, so the while loop just became infinite after the limit is exceeded. 

One solution is the refactor the while loop out into its own method, say "NestedModulesExceedsLimit", and return true instead of continuing, false otherwise. Then...
```csharp
if (!NestedModulesExceedsLimit(...) { 
    files.Add(new AnalysisFile(file, File.ReadAllText(file)));
}
```